### PR TITLE
Is/hw2

### PR DESCRIPTION
**Pull request by SapryginIgor**

### CODE REVIEW

**Comment by yana-tanch** [comment]
- Данные:

1. Неккоретная аугментация
Вы применяете RandomResizedCrop
Объекты в CIFAR-10 уже занимают почти весь кадр (32x32)
Маленький кроп = потеря объекта полностью
(можно посохронять себе картинки с примененной аугментацией и прямо посмотреть на них глазками)

2.  return внутри with
```
def __getitem__(self, index):
    with Image.open(self.files[index]) as image:   
        image = self.transform(image)             
        label = self.labels[index]
        return image, label
```                   

- Архитектура:

1. Обычно Dropout ставят после активаций, а BatchNorm используют до них
BatchNorm уже стабилизирует обучение, внезапно применяемый Dropout мешает нормализации

- Обучение:

1. Обучение происходит всегда на CPU
2. Модель сохраняется в корень проекта

- Тестирование:

1. Нет model.eval()
2. Все на CPU, нет перемещений на device
4. Нет общей точности, только accuracy по классам + непонятный вывод — 10 чисел без объяснения 
5. Сохраняются ошибки только из первых 20 картинок, а не все, и по пути '/content/drive/MyDrive/computer-vision/data/hw2-wrong-preds', которого может вообще не быть, например, при локальном запуске

**Comment by SapryginIgor** [comment]
> * Данные:
> 
> 1. Неккоретная аугментация
>    Вы применяете RandomResizedCrop
>    Объекты в CIFAR-10 уже занимают почти весь кадр (32x32)
>    Маленький кроп = потеря объекта полностью
>    (можно посохронять себе картинки с примененной аугментацией и прямо посмотреть на них глазками)
> 2. return внутри with
> 
> ```
> def __getitem__(self, index):
>     with Image.open(self.files[index]) as image:   
>         image = self.transform(image)             
>         label = self.labels[index]
>         return image, label
> ```
> 
> * Архитектура:
> 
> 1. Обычно Dropout ставят после активаций, а BatchNorm используют до них
>    BatchNorm уже стабилизирует обучение, внезапно применяемый Dropout мешает нормализации
> 
> * Обучение:
> 
> 1. Обучение происходит всегда на CPU
> 2. Модель сохраняется в корень проекта
> 
> * Тестирование:
> 
> 1. Нет model.eval()
> 2. Все на CPU, нет перемещений на device
> 3. Нет общей точности, только accuracy по классам + непонятный вывод — 10 чисел без объяснения
> 4. Сохраняются ошибки только из первых 20 картинок, а не все, и по пути '/content/drive/MyDrive/computer-vision/data/hw2-wrong-preds', которого может вообще не быть, например, при локальном запуске

Все пофиксил, кажется

**Comment by yana-tanch** [comment]
1. Всего 90 картинок сохраняются как ошибочные
Тестовый датасет состоит из 10000 картинок
И при этом accuracy 0.53 

2. Низкое значение точности, нужно улучшать

**Comment by SapryginIgor** [comment]
> 1. Всего 90 картинок сохраняются как ошибочные
>    Тестовый датасет состоит из 10000 картинок
>    И при этом accuracy 0.53
> 2. Низкое значение точности, нужно улучшать

Я пофиксил) надо было убрать все аугментации

**Review by yana-tanch — APPROVED** [review]
